src/oic/oic/consumer.py

```diff
@@ -384,6 +430,12 @@ def parse_authz(self, query="", **kwargs):
                 idt = aresp["id_token"]
             except KeyError:
                 idt = None
+            else:
+                try:
+                    session_update(self.sso_db, _state, "smid", idt["sid"])
+                except KeyError:
```

Comment:
Is this really necessary? If yes, than it needs tests.

Reply:
Done